calico_node/tests/st/config/test_felix_config.py

```diff
@@ -0,0 +1,463 @@
+            result = host.execute("docker exec calico-node ps -a | grep -c calico-felix || exit 0")
+            if result == "1":
+                _log.info("calico-felix restarted.")
+                return
+            else:
+                _log.info("retry [%s] calico-felix waiting for restart", retry)
+                time.sleep(1)
+
+        self.fail("Felix failed to start in 5 seconds.")
+
+    def _get_check_file_func(self, host, expect, log_file, keyword="", snapshot=""):
+        func = partial(self.log_check_file, host, expect, log_file, keyword, snapshot)
+        return func
+
+    def log_check_file(self, host, expect, log_file, keyword="", snapshot=""):
+        _log.info("Check log file %s, keywords %s, expect %s", log_file, keyword, expect)
+        if not snapshot == "":
+            _log.info("Work out the log_file with snapshot")
+            # If snapshot is specified, log_file is a screen log.
+            self.log_screen_diff(host, snapshot, log_file)
+
+        cmd = "test -e %s ; echo $?" % log_file
+        result = host.execute(cmd)
+        if not result == "0":
+            self.fail(("Log file %s not exists." % log_file))
+
+        if keyword != "":
+            cmd = "grep -c %s %s || exit 0" % (keyword, log_file)
+            result = host.execute(cmd, raise_exception_on_failure=False)
+            _log.info("Check keywords result is %s", result)
+            if result == "0" and expect:
+                self.fail(("Log file %s does not contain keyword %s which is not "
+                           "expected." % (log_file, keyword)))
+            if result != "0" and not expect:
+                self.fail(("Log file %s contains keyword %s which is not "
+                           "expected." % (log_file, keyword)))
+
+    def setUp(self):
+        # Override the per-test setUp to avoid wiping etcd; instead only clean up the data we
+        # added.
+        self.remove_host_endpoint()
+
+    def tearDown(self):
+        self.remove_host_endpoint()
+        super(TestFelixConfig, self).tearDown()
+
+    @classmethod
+    def tearDownClass(cls):
+        # Tidy up
+        for host in cls.hosts:
+            host.remove_workloads()
+        for host in cls.hosts:
+            # Just check CRITICAL message because some error messages are
+            # generated by test cases.
+            host.cleanup(err_words={"CRITICAL"})
+            del host
+
+        clear_on_failures()
+
+    def add_host_iface(self, node_name, ip):
+        host_endpoint_data = {
+            'apiVersion': 'projectcalico.org/v3',
+            'kind': 'HostEndpoint',
+            'metadata': {
+                'name': 'host-int',
+                'labels': {'nodeEth': 'host'}
+            },
+            'spec': {
+                'node': node_name,
+                'interfaceName': 'eth0',
+                'expectedIPs': [str(ip)],
+            }
+        }
+        self.host1._apply_resources(host_endpoint_data)
```

Comment:
The function you're calling here has an underscore on the front of the name `_apply_resources`.  That (by convention) means its a private function, so likelihood is that you're supposed to be calling something else. 

Looking at `docker_host.py` I think you should be calling `add_resource()` instead.